Sources/AudioKitUI/Visualizations/FFTView.swift

```diff
@@ -22,38 +22,45 @@ class FFTModel: ObservableObject {
         for index in 0..<fftData.count {
             if fftData[index].isNaN { fftData[index] = 0.0 }
         }
+        
+        var tempAmplitudeArray : [Double] = []
 
         // loop by two through all the fft data
         for i in stride(from: 0, to: FFT_SIZE - 1, by: 2) {
-            // get the real and imaginary parts of the complex number
-            let real = fftData[i]
-            let imaginary = fftData[i + 1]
-            
-            let normalizedBinMagnitude = 2.0 * sqrt(real * real + imaginary * imaginary) / Float(FFT_SIZE)
-            let amplitude = Double(20.0 * log10(normalizedBinMagnitude))
             
-            // scale the resulting data
-            var scaledAmplitude = (amplitude + 250) / 229.80
+            if i / 2 < self.amplitudes.count {
             
-            // restrict the range to 0.0 - 1.0
-            if scaledAmplitude < 0 {
-                scaledAmplitude = 0
-            }
-            if scaledAmplitude > 1.0 {
-                scaledAmplitude = 1.0
-            }
-            
-            // add the amplitude to our array (further scaling array to look good in visualizer)
-            DispatchQueue.main.async {
-                if i / 2 < self.amplitudes.count {
-                    var mappedAmplitude = self.map(n: scaledAmplitude, start1: 0.3, stop1: 0.9, start2: 0.0, stop2: 1.0)
-                    if mappedAmplitude < 0.0 {
-                        mappedAmplitude = 0.0
-                    }
-                    self.amplitudes[i / 2] = mappedAmplitude
+                // get the real and imaginary parts of the complex number
+                let real = fftData[i]
+                let imaginary = fftData[i + 1]
+                
+                let normalizedBinMagnitude = 2.0 * sqrt(real * real + imaginary * imaginary) / Float(FFT_SIZE)
+                let amplitude = Double(20.0 * log10(normalizedBinMagnitude))
+                
+                // scale the resulting data
+                var scaledAmplitude = (amplitude + 250) / 229.80
+                
+                // restrict the range to 0.0 - 1.0
+                if scaledAmplitude < 0 {
+                    scaledAmplitude = 0
+                }
+                if scaledAmplitude > 1.0 {
+                    scaledAmplitude = 1.0
                 }
+                    
+                // further scaling array to look good in visualizer
+                var mappedAmplitude = self.map(n: scaledAmplitude, start1: 0.3, stop1: 0.9, start2: 0.0, stop2: 1.0)
```

Comment:
Same comment for these numbers here, 0.3 & 0.9 - just wondering if they could/should be parameterised or calculated somehow.

Reply:
At the very least, you would think that there should be no need to have multiple lines to map the data appropriately even if an arbitrary scaling of the output data is required. I'll revisit this and see if I can sort out the magic numbers.

Reply:
Thanks for noting it. None of this is show stopping - just comments I can make from afar.
Again I really appreciate you doing this.

Reply:
I looked into this and I've cleared as much of the magic away as I could. Check out the following pull requests:
https://github.com/AudioKit/AudioKit/pull/2381
https://github.com/AudioKit/AudioKitUI/pull/7

Now, FFTView just has one map function which takes the expected amplitude values and maps them to a 0-1 range for the visualizer. The amplitude values matched what I was seeing in the spectrum analyzer in Ableton Live, so I'm content with it for now.

As far as FFTTap goes, it needed a scaling factor on it's output when it was not normalizing the fft data. Apple's documentation brings up this need for a scaling factor with their fft.